persistence/genesis.go

```diff
@@ -44,97 +45,72 @@ func (m *persistenceModule) populateGenesisState(state modules.PersistenceGenesi
-		}
-		if err = addValueToPool(types.PoolNames_AppStakePool.String(), act.GetStakedAmount()); err != nil {
-			log.Fatalf("an error occurred inserting staked tokens into %s pool: %s", types.PoolNames_AppStakePool, err.Error())
-		}
-	}
-	for _, act := range state.GetNodes() {
-		addrBz, err := hex.DecodeString(act.GetAddress())
-		if err != nil {
-			log.Fatalf("an error occurred converting address to bytes %s", act.GetAddress())
-		}
-		pubKeyBz, err := hex.DecodeString(act.GetPublicKey())
-		if err != nil {
-			log.Fatalf("an error occurred converting pubKey to bytes %s", act.GetPublicKey())
-		}
-		outputBz, err := hex.DecodeString(act.GetOutput())
-		if err != nil {
-			log.Fatalf("an error occurred converting output to bytes %s", act.GetOutput())
-		}
-		err = rwContext.InsertServiceNode(addrBz, pubKeyBz, outputBz, false, StakedStatus, act.GetGenericParam(), act.GetStakedAmount(), act.GetChains(), act.GetPausedHeight(), act.GetUnstakingHeight())
-		if err != nil {
-			log.Fatalf("an error occurred inserting a service node in the genesis state: %s", err.Error())
-		}
-		if err = addValueToPool(types.PoolNames_ServiceNodeStakePool.String(), act.GetStakedAmount()); err != nil {
-			log.Fatalf("an error occurred inserting staked tokens into %s pool: %s", types.PoolNames_ServiceNodeStakePool.String(), err.Error())
-		}
-	}
-	for _, act := range state.GetFish() {
-		addrBz, err := hex.DecodeString(act.GetAddress())
-		if err != nil {
-			log.Fatalf("an error occurred converting address to bytes %s", act.GetAddress())
-		}
-		pubKeyBz, err := hex.DecodeString(act.GetPublicKey())
-		if err != nil {
-			log.Fatalf("an error occurred converting pubKey to bytes %s", act.GetPublicKey())
-		}
-		outputBz, err := hex.DecodeString(act.GetOutput())
-		if err != nil {
-			log.Fatalf("an error occurred converting output to bytes %s", act.GetOutput())
-		}
-		err = rwContext.InsertFisherman(addrBz, pubKeyBz, outputBz, false, StakedStatus, act.GetGenericParam(), act.GetStakedAmount(), act.GetChains(), act.GetPausedHeight(), act.GetUnstakingHeight())
-		if err != nil {
-			log.Fatalf("an error occurred inserting a fisherman in the genesis state: %s", err.Error())
-		}
-		if err = addValueToPool(types.PoolNames_FishermanStakePool.String(), act.GetStakedAmount()); err != nil {
-			log.Fatalf("an error occurred inserting staked tokens into %s pool: %s", types.PoolNames_FishermanStakePool.String(), err.Error())
-		}
+
+	stakedActorsInsertConfigs := []struct {
+		Name     string
+		Getter   func() []*coreTypes.Actor
+		InsertFn func(address []byte, publicKey []byte, output []byte, paused bool, status int32, serviceURL string, stakedTokens string, chains []string, pausedHeight int64, unstakingHeight int64) error
+		Pool     coreTypes.PoolNames
+	}{
+		{
+			Name:     "app",
+			Getter:   state.GetApplications,
+			InsertFn: rwContext.InsertApp,
+			Pool:     coreTypes.PoolNames_POOL_NAMES_APP_STAKE_POOL,
+		},
+		{
+			Name:     "node",
```

Comment:
```suggestion
			Name:     "serviceNode",
```

TOL(Thinking Out Loud): I'm beginning to think we should rename these actors to `Servicers`

Reply:
I agree, serviceNode is an improvement here since node could mean anything. If we want to rename them to Servicers I guess the only thing I would do is make sure that we are consistent in the codebase, documentation and perhaps even in Notion. Thoughts?